### DISCUSSION TITLE
M2M relationship in both objects

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#14052] [Archive]
Hi!

Not sure if this is even possible, or if I'll go into endless loops or anything. But I have a self referencing M2M relationship. The articles collection had an M2M field called "related_articles" referencing the articles collection itself.

Now, when I add a related article to an article, I would also like to automatically set the relation the other way around. Is that a feature at the moment? Or should I go with a hook on the junction table?

Cheers!

Kees.

**Comment by rijkvanzanten** [comment]
> Now, when I add a related article to an article, I would also like to automatically set the relation the other way around.

What do you mean by this exactly? If you add a related article to an article in a m2m field, you create a new record in the junction table which links the two. Seeing the table is self referencing, this should mean that the record will show up on both articles 🤔

**Reply by keesvanbemmel** [reply]
Well, to be frank, I think I'm missing or misinterpreting something.

I was under the impression that if I add related articles, the relation would also show up in the details of those related articles. That doesn't seem to be the case as this screen recording shows:

https://user-images.githubusercontent.com/4633876/175354653-e0f6c807-d6de-4a48-b679-82d400febe8f.mov

**Reply by rijkvanzanten** [reply]
Ah you're right! Can confirm that I'm seeing the same behavior here.

I believe this is because the m2m interface is currently using a filter that only checks against the foreign key field that points to the "related" item, rather than both the current and related item in the junction table. This makes sense for most relationships, as the other key wouldn't exist in the current collection. That being said, for self-referencing relationships to the same collection, it should look at both foreign keys, as it shouldn't matter in which of the two the current item resides.

Actually.. on second thought, we might want to make that configurable 🤔 I can imagine there's use case where you'd use self-referencing as a tree-path, rather than a circular dependency. For "related articles" it makes sense to have them "auto-linked" from both ways, but for a "continue reading", you might not want it to go "backwards" at all 🤔 (cc @benhaynes)

**Reply by keesvanbemmel** [reply]
And I was sure I was missing something! 👯 

Thanks for getting back to me. I'll mark this question as answered and create an "issue" out of it for a new feature somewhere along the line ok? Feel free to close the issue instantly if you don't want it on your board!

edit: a feature request just reverts me to this discussion board so I'll live it like this :)

**Reply by rijkvanzanten** [reply]
Lets open a new discussion as a feature request though 😄 I'd love to hear some more comments from the community about whether or not I'm going insane in assuming there's a use case for both configurations 🤔

**Reply by keesvanbemmel** [reply]
Will do!

**Reply by benhaynes** [reply]
@rijkvanzanten — let's discuss over video so I can fully wrap my head around this. 🧠
